scripts/WFCORE-5261-common-conf.adoc

```diff
@@ -0,0 +1,104 @@
+= Add Common Script Configuration File
+:author:            James R. Perkins
+:email:             jperkins@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+There is a desire to have a common script configuration file, `common.conf`. This will be called before the standalone
+or domain script configuration file. The other scripts will also invoke the common script configuration file. It allows
+things like common environment variables to be set by users regardless of the script being used.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFCORE-5261[WFCORE-5261]
+* https://issues.redhat.com/browse/WFLY-14399[WFLY-14399]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/EAP7-1251[EAP7-1251]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:spriadka@redhat.com[Simon Priadka]
+
+=== Testing By
+
+* [x] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+* WildFly Core
+* WildFly
+* Any project or component which invokes script files
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [x] Managed domain
+
+* [x] OpenShift s2i
```

Comment:
Do we really want to expose custom scripts into runtime? So far we have only custom s2i scripts in build time (that is safe). But introducing custom scripts into runtime open whole new set of problems. As immutability of image. Consistency of EAP pods in cluster. When will be change applied? ...

Also in Openshift we have native way for setting environment variables - ConfigMaps. I think we should recommend this and not come with own proprietary solution.

I read the linked customer report and it was not specifically targeted for OpenShift environment.

Can we avoid advertising this feature in OpenShift for customers?

@jmesnil @luck3y WDYT?

Reply:
We essentially already do this with the `standalone.conf`. The only difference between that and the `common.conf` is where in the `standalone.sh` it runs and the fact that it will run on all scripts. I only marked it here because I thought the end user would be able to add it. However I don't think it's something we should provide in the image so maybe I'm wrong.